sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2663,6 +2663,24 @@ def check_resources_fit_cluster(
                             'stores, but the existing cluster with '
                             f'{launched_resources!r} does not support FUSE '
                             f'mounting. Launch a new cluster to run this task.')
+                if (example_resource.accelerators is not None and
+                        launched_resources.accelerators is not None):
+                    for acc in example_resource.accelerators:
+                        if acc not in launched_resources.accelerators:
+                            continue
+                        self_count = example_resource.accelerators[acc]
+                        existing_count = launched_resources.accelerators[acc]
+                        if (isinstance(existing_count, float) and
+                                not existing_count.is_integer() and
+                                not math.isclose(self_count, existing_count)):
+                            with ux_utils.print_exception_no_traceback():
+                                raise exceptions.ResourcesMismatchError(
+                                    'Task requested resources with fractional '
+                                    'accelerator counts. For fractional '
+                                    'counts, the required count must match the '
+                                    'existing cluster. Got required accelerator'
+                                    f' {acc}:{self_count} but the existing '
+                                    f'cluster has {acc}:{existing_count}.')
```

Comment:
This error message is not accurate? Our check is for ACC count of existing cluster instead of the task requested resources?

Reply:
Please see the above comments 🤔